plugins/woocommerce/client/admin/client/settings-payments/components/official-badge/official-badge.tsx

```diff
@@ -0,0 +1,107 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { __ } from '@wordpress/i18n';
+import { Popover } from '@wordpress/components';
+import { Link, Pill } from '@woocommerce/components';
+import { createInterpolateElement, useState } from '@wordpress/element';
+import { Icon } from '@wordpress/icons';
+import { useDebounce } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import { BadgeOfficial } from '../icons';
+
+interface OfficialBadgeProps {
+	/**
+	 * The style of the badge.
+	 */
+	variant: 'expanded' | 'compact';
+}
+
+/**
+ * A component that displays an official badge.
+ * The style of the badge can be either "expanded" or "compact".
+ *
+ * @example
+ * // Render an official badge with icon and text.
+ * <OfficialBadge variant="expanded" />
+ *
+ * @example
+ * // Render an official badge with just the icon.
+ * <OfficialBadge variant="compact" />
+ */
+export const OfficialBadge = ( { variant }: OfficialBadgeProps ) => {
+	const [ isPopoverVisible, setPopoverVisible ] = useState( false );
+
+	const hidePopoverDebounced = useDebounce( () => {
+		setPopoverVisible( false );
+	}, 350 );
+	const showPopover = () => {
+		setPopoverVisible( true );
+		hidePopoverDebounced.cancel();
+	};
+
+	return (
+		<Pill className={ `woocommerce-official-extension-badge` }>
+			<span
+				className="woocommerce-official-extension-badge__container"
+				tabIndex={ 0 }
+				role="button"
+				onClick={ () => setPopoverVisible( ! isPopoverVisible ) }
+				onMouseEnter={ showPopover }
+				onMouseLeave={ hidePopoverDebounced }
+				onKeyDown={ ( event ) => {
+					if ( event.key === 'Enter' || event.key === ' ' ) {
+						setPopoverVisible( ! isPopoverVisible );
+					}
+				} }
+			>
+				<Icon size={ 16 } icon={ BadgeOfficial } />
+				{ variant === 'expanded' && (
+					<span>{ __( 'Official', 'woocommerce' ) }</span>
+				) }
+				{ isPopoverVisible && (
+					<Popover
+						className="woocommerce-official-extension-badge-popover"
+						placement="top-start"
+						offset={ 4 }
+						variant="unstyled"
+						focusOnMount={ true }
+						noArrow={ true }
+						shift={ true }
+						onClose={ hidePopoverDebounced }
+					>
+						<div className="components-popover__content-container">
+							<p>
+								{ createInterpolateElement(
+									__(
+										'This is an Official WooCommerce payment extension. <learnMoreLink />',
```

Comment:
should `Official` start from capital O?

Reply:
That is how it is in the designs. Right @orcungogus?

Reply:
Looks right to me, I guess we want `Official WooCommerce` to be like a signifier of a branded, trusted plugin.